src/vue/common/CollectionLoader.spec.js

```diff
@@ -0,0 +1,211 @@
+    data: [],
+    fetchNext: () => {}
+  })
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('loadPage calls provided loaderFn', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const spy = sinon.stub().resolves({
+      data: [],
+      fetchNext: () => {}
+    })
+
+    await wrapper.vm.loadPage('event', spy)
+
+    expect(spy.calledOnce).to.be.true
+  })
+
+  it('loadPage properly emits event with correct payload', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const data = []
+    const event = 'event'
+    await wrapper.vm.loadPage(event, sinon.stub().resolves({
+      data
+    }))
+    expect(TestHelper.isEmittedOnce(wrapper, event, data)).to.be.true
+  })
+
+  it('loadPage saves fetchNext returned in response', async () => {
+    const fetchNext = () => {}
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+
+    const spy = sinon.stub().resolves({
+      fetchNext,
+      data: []
+    })
+
+    await wrapper.vm.loadPage('event', spy)
+
+    expect(wrapper.vm.nextPageLoader).to.equal(fetchNext)
+  })
+
+  it('loadPage sets isCollectionFetched to false when data is not fully loaded',
+    async () => {
+      const fetchNext = () => {}
+      const data = []
+      const pageLimit = 10
+      const wrapper = shallowMount(CollectionLoader, {
+        propsData: {
+          firstPageLoader: sinon.stub().resolves({
+            data,
+            fetchNext
+          }),
+          pageLimit
+        },
+        localVue
+      })
+      data.length = pageLimit + 1
+      const spy = sinon.stub().resolves({
+        fetchNext,
+        data
+      })
+
+      await wrapper.vm.loadPage('event', spy)
+
+      expect(wrapper.vm.isCollectionFetched).to.be.false
+    })
+
+  it('loadPage sets isCollectionFetched to true when data is fully loaded',
+    async () => {
+      const fetchNext = () => {}
+      const data = []
+      const pageLimit = 10
+      const wrapper = shallowMount(CollectionLoader, {
+        propsData: {
+          firstPageLoader: sinon.stub().resolves({
+            data,
+            fetchNext
+          }),
+          pageLimit
+        },
+        localVue
+      })
+      data.length = pageLimit - 1
+      const spy = sinon.stub().resolves({
+        fetchNext,
+        data
+      })
+
+      await wrapper.vm.loadPage('event', spy)
+
+      expect(wrapper.vm.isCollectionFetched).to.be.true
+    })
+
+  it('loadPage delegates error to ErrorHandler when got one', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const error = new Error('Error')
+    const spy = sinon.stub().throws(error)
```

Comment:
In the tests it’s more preferable to use adjectives in function and variable names. Also in the current case we got some reading-complicated points:
- actually we store stub in the variable named `spy`. It misleads the reader
- when we define a variable with name containing `spy` or `stub` the reader expects using of spy- or stub-related methods. in the cases when we do not need spy/stub-related methods, we shouldn’t name the variable with such names. better to name it `func` or `fn`

I’d recommend renaming the variable to errorThrownFn cuz (1) it explains what the function does — it throws an error, (2) despite the function is actually a stub we do not need any of stub’s methods, the only thing we need here is the function that throws something, we do not need any assertions, so it contains `fn` in the name

The best tests are the tests that are easy understandable without ready any of the tested code. We